site/content@beta/backend-apis/key-concepts.md

```diff
@@ -0,0 +1,127 @@
+
+A **message** is not the only event type, though. To check out other event types, see the [Events](#events) section.
+
+
+### Rules and conditions
+
+Here are some general rules, which summarize the previous section and add new info.
+
+1. When a new chat is started, **a new active thread** is created within this chat. New threads within a single chat are created on the server side. 
+
+2. There's always **only one active thread**. Only the **last thread** can be the **active** one. Events are always added to the active thread. 
+
+3. There can be time gaps between threads in a chat, but once a chat is started, it's **continuously open**.
+
+4. Messages are sent and delivered even when the recipient (both Customer and Agent) is **offline**.
+
+5. **Multiple Agents** can participate in a single chat. 
+
+6. Every user can have **multiple concurrent chats**. Read how it applies to [Agents](#chatting-as-an-agent) and [Customers](#chatting-as-a-customer).
+
+7. The algorithm that decides how chats are distributed between Agents is called **routing**. It's documented in the [Routing](#routing) section.
+
+
+### The flow 
+
+To better understand the flow between services, watch a short video.
+
+<!-- Animacja do zmiany. Zajmie sie tym Gosia oraz Bartek. -->
+
+<video loop width="750" height="500" style="height: auto;" controls>
+<source type="video/mp4" src="/beta-docs/platform-overview/images/simple_event_schema.mp4">
+</video>
+
+
+## Chat routing
+
+Routing is the process of **assigning chats to agents**. The primary goal of the routing mechanism is to distribute chats to all available agents who use the same license.  
+<!-- In the future, it will be possible to create a custom routing mechanism and run it on the LiveChat platform. -->
+
+Chats are being assigned to agents **either automatically or manually**. Our default routing mechanisms are described in [Understanding chat routing](https://www.livechatinc.com/help/understanding-chat-routing/).
+
+<!-- ### Automatic routing
+
+The diagram below present the flow of automatic routing.
+
+![Automatic routing](automatic-routing.png "automatic routing")
+
+### Manual routing
+
+The diagram below present the flow of manual routing.
+
+![Manual routing](manual-routing.png "manual routing") -->
+
+<!-- 1. Czy dajemy te grafiki, przerabiamy je? 
+
+    2. Czy wspominamy o custom routingu, ktory bedzie mozliwy w przyszlosci? 
+    
+    3. Czy rezygnujemy z opisow i linkujemy do artykulu w Help Center? 
+    
+    4. Fajnie by bylo tu kiedys nagrac filmik z lektorem  -->
+
+#### Router system messages
+
+While the chat changes its state, the router sends system messages. Read more about **system messages** in [Agent](#agent-system-messages) and [Customer](#customer-system-messages) Chat API.
+
+<!-- Nie wiem trochę o co chodzi z tymi rodzajami system messages. Czy z grubsza mozna je podzielić na takie po stronie Agenta i po stronie Customera. Wtedy mona by tu wymienic typy
```

Comment:
I don't think this doc is the right place to present examples of system messages. I'm not even sure if we should list them here. Isn't it too detailed?